test/fixtures.js

```diff
@@ -0,0 +1,175 @@
+const { ethers } = require("hardhat");
+const {
+  getDecimals,
+  getTokenAmount,
+} = require("../lib/helpers/conversions.js");
+
+const {
+  utils: { parseEther, parseUnits },
+} = ethers;
+
+const { time } = require("@openzeppelin/test-helpers");
+
+let softCap;
+let hardCap;
+let price;
+let startTime;
+let endTime;
+
+// constants
+const sevenDaysInSeconds = time.duration.days(7);
+const tenDaysInSeconds = time.duration.days(10);
+const hundredDaysInSeconds = time.duration.days(100);
+const twentySeconds = time.duration.seconds(20);
+
+async function launchFixture() {
```

Comment:
Such long functions might be a coding style you agreed upon, which is also a valid one.
So view the following comment as _optional_

Personally, because this function does a lot of preparing, I would have
- split each "section" (indicated by these long "------" separators) into its own function
- (this long function might have been the reason for the "spaghetti" test file in the first place)
- reduce the amount of "duplicate comments" (eg. the comments reads "Seed Factory" and the variable also does)

Reply:
Compare "Builder pattern"

https://github.com/PrimeDAO/prime-deals-dapp/blob/8b236a3c7d4ae427bc02066308af7f143fe77805/cypress/fixtures/dealFixtures.ts#L128